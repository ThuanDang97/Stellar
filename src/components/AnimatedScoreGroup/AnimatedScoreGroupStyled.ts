import styled, { keyframes } from 'styled-components'

// Components
import { TextStyled, ViewStyled } from '@components/styled-components'

// Utils
import { pxToRem } from '@utils/theme'

// Props type
import { AnimatedWrapperProps } from '@self-types/components/Animation.props'

// Theme
import Colors from '@themes/Colors'
import { TitleStyled } from '@components/Title/TitleStyled'
import { TextProps } from '@self-types/components/Text.props'

const slide = keyframes`
to {
  transform: translateX(-100%)
}
`

const fadeIn = keyframes`
from {
  opacity: 0
}
to {
  opacity: 1
}
`

const fadeOut = keyframes`
from {
  opacity: 1
}
to {
  opacity: 0
}
`

const fadeAndSlideIn = keyframes`
  0% {
    opacity: 0;
    transform: translateX(${pxToRem(20)})
  }
  60%, 100% {
    opacity: 1;
    transform: translateX(0)
  }
`

const changeTextColor = keyframes`
to {
  color: ${Colors.caribbeanGreen}
}
`

// Title
export const TitleWrapperStyled = styled(ViewStyled)<AnimatedWrapperProps>`
  display: flex;
  width: 80%;
  margin-left: 10%;
  margin-bottom: ${({ theme }) => pxToRem(theme.metrics.dimensions.xxl)};

  overflow: hidden;

  ${TextStyled} {
    flex: 0 0 100%;

    &:nth-child(1) {
      animation: ${fadeOut} 0.4s linear both
          ${({ delay }) => `${(delay ?? 0) + 7.8}s`},
        ${slide} 0s linear forwards ${({ delay }) => `${(delay ?? 0) + 8.2}s`};
    }

    &:nth-child(2) {
      animation: ${slide} 0s linear both
          ${({ delay }) => `${(delay ?? 0) + 7.75}s`},
        ${fadeIn} 1s ease-out both ${({ delay }) => `${(delay ?? 0) + 7.8}s`};
    }
  }
`

// Labels
export const LabelWrapperStyled = styled(ViewStyled)<AnimatedWrapperProps>`
  display: flex;
  margin-top: ${({ theme }) => pxToRem(theme.metrics.dimensions.lg)};
  gap: ${({ theme }) => pxToRem(theme.metrics.dimensions.sm)};

  /* Set color of plus icons always white */
  & > [data-icon='true'] {
    color: ${({ theme }) => theme.colors.white};
  }

  ${TextStyled} {
    &:nth-child(1) {
      animation: ${fadeAndSlideIn} 0.6s both
          ${({ delay }) => `${(delay ?? 0) + 0.5}s`},
        ${changeTextColor} 1s forwards
          ${({ delay }) => `${(delay ?? 0) + 7.8}s`};
    }

    &:nth-child(2) {
      animation: ${fadeAndSlideIn} 0.6s both
        ${({ delay }) => `${(delay ?? 0) + 1.3}s`};
    }
    &:nth-child(3) {
      animation: ${fadeAndSlideIn} 0.6s both
          ${({ delay }) => `${(delay ?? 0) + 1.3}s`},
        ${changeTextColor} 1s forwards
          ${({ delay }) => `${(delay ?? 0) + 7.8}s`};
    }

    &:nth-child(4) {
      animation: ${fadeAndSlideIn} 0.6s both
        ${({ delay }) => `${(delay ?? 0) + 2.1}s`};
    }
    &:nth-child(5) {
      animation: ${fadeAndSlideIn} 0.6s both
          ${({ delay }) => `${(delay ?? 0) + 2.1}s`},
        ${changeTextColor} 1s forwards
          ${({ delay }) => `${(delay ?? 0) + 7.8}s`};
    }

    &:nth-child(6),
    &:nth-child(7) {
      animation: ${fadeAndSlideIn} 0.6s both
        ${({ delay }) => `${(delay ?? 0) + 2.9}s`};
    }

    &:nth-child(8),
    &:nth-child(9) {
      animation: ${fadeAndSlideIn} 0.6s both
        ${({ delay }) => `${(delay ?? 0) + 3.7}s`};
    }
  }
`

export const ScoreLabelStyled = styled(TitleStyled)<TextProps>`
  font-size: ${({ theme }) => pxToRem(theme.typography.fontSize.sm)};
  line-height: ${({ theme }) => pxToRem(theme.typography.lineHeight.md)};

  @media (max-width: ${({ theme }) => pxToRem(theme.metrics.breakPoints.xs)}) {
    font-size: ${({ theme }) => pxToRem(theme.typography.fontSize.base)};
    line-height: ${({ theme }) => pxToRem(theme.typography.lineHeight.md)};
  }
`
