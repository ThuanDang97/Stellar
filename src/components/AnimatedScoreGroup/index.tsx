import React, { memo } from 'react'
import { useTheme } from 'styled-components'

// Constants
import { SCORE_LABELS, ANIMATION_SCORE_TITLE } from '@constants/index'

// Components
import AnimatedLineScore from '@components/AnimatedLineScore'
import { ViewStyled } from '@components/styled-components'

// Props type
import { AnimationComponentProps } from '@self-types/components/Animation.props'

// Utils
import { pxToRem } from '@utils/theme'
import {
  LabelWrapperStyled,
  ScoreLabelStyled,
  TitleWrapperStyled,
} from './AnimatedScoreGroupStyled'
import { TitleLandingPageStyled } from '../../../styles/pages/LandingPageStyled'

const AnimatedScoreGroup = ({ delay }: AnimationComponentProps) => {
  const theme = useTheme()

  return (
    <ViewStyled
      width={theme.metrics.width.full}
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
    >
      <ViewStyled width={`min(90%, ${pxToRem(700)})`}>
        {/* Title */}
        <TitleWrapperStyled delay={delay}>
          {ANIMATION_SCORE_TITLE.map((title) => (
            <TitleLandingPageStyled
              key={title}
              fontWeight={theme.typography.fontWeight.bold}
              color={theme.colors.white}
              as="h1"
            >
              {title}
            </TitleLandingPageStyled>
          ))}
        </TitleWrapperStyled>

        {/* Animated score */}
        <AnimatedLineScore delay={delay} />
      </ViewStyled>

      {/* Labels */}
      <LabelWrapperStyled delay={delay}>
        {SCORE_LABELS.map((label, index) => (
          <ScoreLabelStyled
            key={`${label + index}`}
            data-icon={label === '+' ? 'true' : 'false'}
            color={theme.colors.white}
            letterSpacing={theme.typography.letterSpacing.xxl}
          >
            {label}
          </ScoreLabelStyled>
        ))}
      </LabelWrapperStyled>
    </ViewStyled>
  )
}

export default memo(AnimatedScoreGroup)
