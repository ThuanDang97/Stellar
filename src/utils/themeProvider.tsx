import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
