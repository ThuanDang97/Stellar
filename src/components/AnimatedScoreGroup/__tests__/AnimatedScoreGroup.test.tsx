// Components
import AnimatedScoreGroup from '@components/AnimatedScoreGroup'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('AnimatedScoreGroup Component', () => {
  it('renders animatedScoreGroup unchanged', () => {
    const { asFragment } = renderWithTheme(<AnimatedScoreGroup />)

    expect(asFragment()).toMatchSnapshot()
  })
})
