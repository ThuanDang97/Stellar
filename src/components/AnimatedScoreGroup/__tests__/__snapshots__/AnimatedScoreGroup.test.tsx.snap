// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedScoreGroup Component renders animatedScoreGroup unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 datfMp"
    display="flex"
    width="100%"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 kQEvr"
      width="min(90%, 43.75rem)"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 AnimatedScoreGroupStyled__TitleWrapperStyled-sc-jka6cd-0 fTsvHV kIwIuJ"
      >
        <h1
          class="TextStyled-sc-9xia41-0 LandingPageStyled__TitleLandingPageStyled-sc-19vluzy-2 hMnkWB ekfgDB"
          color="#FFF"
          font-weight="700"
        >
          Your credit score is calculated using data from your credit report.
        </h1>
        <h1
          class="TextStyled-sc-9xia41-0 LandingPageStyled__TitleLandingPageStyled-sc-19vluzy-2 hMnkWB ekfgDB"
          color="#FFF"
          font-weight="700"
        >
          Stellar improves factors that make up 80% of your score.
        </h1>
      </div>
      <div
        class="ViewStyled-sc-rqe5sa-0 kCJugR"
        display="flex"
        overflow="hidden"
        width="100%"
      >
        <div
          class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__StartBaseLineStyled-sc-okjczv-1 AnimatedLineScoreStyled__StartLineScoreStyled-sc-poybgv-0 dlHdcM hLygIM dPidGS"
          width="35"
        />
        <div
          class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__MiddleBaseLineStyled-sc-okjczv-2 AnimatedLineScoreStyled__MiddleLineScoreStyled-sc-poybgv-1 btYAst ldvwuf fvdruU"
          width="30"
        />
        <div
          class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__MiddleBaseLineStyled-sc-okjczv-2 AnimatedLineScoreStyled__MiddleLineScoreStyled-sc-poybgv-1 dCqiri ldvwuf hkQYyp"
          width="15"
        />
        <div
          class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__MiddleBaseLineStyled-sc-okjczv-2 AnimatedLineScoreStyled__MiddleLineScoreStyled-sc-poybgv-1 hhHjIH ldvwuf ifzRSN"
          width="10"
        />
        <div
          class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__EndBaseLineStyled-sc-okjczv-3 AnimatedLineScoreStyled__EndLineScoreStyled-sc-poybgv-2 hhHjIH dahDgB jlmGRx"
          width="10"
        />
        <div
          class="LineStyled__BaseLineStyled-sc-okjczv-0 LineStyled__StartBaseLineStyled-sc-okjczv-1 AnimatedLineScoreStyled__LayerLineStyled-sc-poybgv-3 kdqgjs hLygIM gwDPoj"
          width="80% - 0.875rem"
        />
        <div
          class="AnimatedLineScoreStyled__Shim-sc-poybgv-4 lisOCr"
        />
        <div
          class="AnimatedLineScoreStyled__Shim-sc-poybgv-4 bDcpsq"
        />
        <div
          class="AnimatedLineScoreStyled__Shim-sc-poybgv-4 cRdoku"
        />
        <div
          class="AnimatedLineScoreStyled__Shim-sc-poybgv-4 dnSNqt"
        />
      </div>
    </div>
    <div
      class="ViewStyled-sc-rqe5sa-0 AnimatedScoreGroupStyled__LabelWrapperStyled-sc-jka6cd-1 fTsvHV iVZsDV"
    >
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="false"
        letter-spacing="0.9"
      >
        35% Payment History
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="true"
        letter-spacing="0.9"
      >
        +
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="false"
        letter-spacing="0.9"
      >
        30% Credit Owed
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="true"
        letter-spacing="0.9"
      >
        +
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="false"
        letter-spacing="0.9"
      >
        15% Credit Age
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="true"
        letter-spacing="0.9"
      >
        +
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="false"
        letter-spacing="0.9"
      >
        10% New Credit
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="true"
        letter-spacing="0.9"
      >
        +
      </span>
      <span
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 AnimatedScoreGroupStyled__ScoreLabelStyled-sc-jka6cd-2 fWhMws ftzJlA kwHUqK"
        color="#FFF"
        data-icon="false"
        letter-spacing="0.9"
      >
        10% Credit Mix
      </span>
    </div>
  </div>
</DocumentFragment>
`;
